pkg/puller/puller.go

```diff
@@ -19,19 +19,31 @@ import (
 	"github.com/ethersphere/bee/pkg/topology"
 )
 
+const defaultShallowBinPeers = 2
+
 var (
-	bins = uint8(16)
+	bins = swarm.MaxBins
 
 	// how many peers per bin do we want to sync with outside of depth
-	shallowBinPeers = 2
-	logMore         = false // enable this to get more logging
+	shallowBinPeers = defaultShallowBinPeers
+
+	logMore = false // enable this to get more logging
 )
 
 type Options struct {
-	StateStore storage.StateStorer
-	Topology   topology.Driver
-	PullSync   pullsync.Interface
-	Logger     logging.Logger
+	StateStore      storage.StateStorer
+	Topology        topology.Driver
+	PullSync        pullsync.Interface
+	Logger          logging.Logger
+	Bins            uint8
+	ShallowBinPeers int
+}
+
+func NewOptions() Options {
+	return Options{
+		Bins:            swarm.MaxBins,
+		ShallowBinPeers: defaultShallowBinPeers,
+	}
```

Comment:
I am not sure if NewOptions is needed. Cold we just set Bins and ShallowBinPeers to their default values in New function if they are set to 0 or less? This would remove the need for constructor for options for another constructor.